Address various warnings as errors

JakeStevens opened the pull request:
Summary:
Some projects uses more restrictive build options than currently used in ET CI.

This means we encountered a number of errors when enabling for a microcontroller.

Reviewed By: digantdesai, swolchok

Differential Revision: D69139962

JakeStevens commented:
https://www.internalfb.com/diff/D69139962

mergennachin commented:
@JakeStevens 

A few remaining errors in OSS CI: https://github.com/pytorch/executorch/pull/8581

and also lint error: 

https://github.com/pytorch/executorch/actions/runs/13422197104/job/37540247065?pr=8581

JakeStevens commented:
@pytorchbot label "topic: not user facing

mergennachin commented:
Almost there.... 

https://github.com/pytorch/executorch/actions/runs/13575560758/job/37950830208?pr=8581

JakeStevens committed:
Address various warnings as errors (#8581)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/8581

Some projects uses more restrictive build options than currently used in ET CI.

This means we encountered a number of errors when enabling for a microcontroller.

Reviewed By: digantdesai, swolchok

Differential Revision: D69139962

mergennachin reviewed (approved):
Thank you so much!
